src/textual/css/stylesheet.py

```diff
@@ -304,10 +306,20 @@ def parse(self) -> None:
         """
         rules: list[RuleSet] = []
         add_rules = rules.extend
+
         for path, (css, is_default_rules, tie_breaker) in self.source.items():
-            css_rules = self._parse_rules(
-                css, path, is_default_rules=is_default_rules, tie_breaker=tie_breaker
-            )
+            if css in self._invalid_css:
+                continue
+            try:
+                css_rules = self._parse_rules(
+                    css,
+                    path,
+                    is_default_rules=is_default_rules,
+                    tie_breaker=tie_breaker,
+                )
+            except Exception:
+                self._invalid_css.add(css)
+                raise
```

Comment:
Is this equivalent to the alternative below?

```py
except Exception as error:
    ...
    raise error from None
```

Reply:
I'm not entirely sure. Doesn't impact anything here since we never display the traceback.